fe/fe-core/src/main/java/org/apache/doris/nereids/rules/expression/rewrite/rules/BetweenToCompoundRule.java

```diff
@@ -0,0 +1,61 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.nereids.rules.expression.rewrite.rules;
+
+import org.apache.doris.nereids.rules.expression.rewrite.AbstractExpressionRewriteRule;
+import org.apache.doris.nereids.rules.expression.rewrite.ExpressionRewriteContext;
+import org.apache.doris.nereids.trees.NodeType;
+import org.apache.doris.nereids.trees.expressions.Between;
+import org.apache.doris.nereids.trees.expressions.CompoundPredicate;
+import org.apache.doris.nereids.trees.expressions.Expression;
+import org.apache.doris.nereids.trees.expressions.GreaterThan;
+import org.apache.doris.nereids.trees.expressions.GreaterThanEqual;
+import org.apache.doris.nereids.trees.expressions.LessThan;
+import org.apache.doris.nereids.trees.expressions.LessThanEqual;
+import org.apache.doris.nereids.trees.expressions.Not;
+
+/**
+ * Rewrites BetweenPredicates into an equivalent conjunctive/disjunctive
+ * CompoundPredicate.
+ * Examples:
+ * A BETWEEN X AND Y ==> A >= X AND A <= Y
+ * A NOT BETWEEN X AND Y ==> A < X OR A > Y
+ */
+public class BetweenToCompoundRule extends AbstractExpressionRewriteRule {
+
+    public static BetweenToCompoundRule INSTANCE = new BetweenToCompoundRule();
+
+    @Override
+    public Expression visitNot(Not expr, ExpressionRewriteContext context) {
```

Comment:
I thins not between should be rewrite in `SimplifyNotExprRule`

Reply:
In the case `Not(a between b and c)`, we can
1. rewrite `a between b and c`  to ` a >= b and a <= c` in `BetweenToCompoundRue`
2. then apply `SimplifyNotExprRule` to rewrite `Not(a >= b and a <= c)` to `Not(a >= b) Or Not(a <= c)` by the De Morgan's laws
3. finally `SimplifyNotExprRule` recursive rewrite `Not(a >= b)` to `a < b`, and `Not(a <= c)` to `a > c` the result is `(a < b) Or (a > c)`.

This steps make the functions independent of each other.

Reply:
you're right

Reply:
In my opinion, SimplifyNotExprRule should be a part of NormalizeExpressionRule. And NormalizeExpressionRule can process not only binary expression, it can process all normalization of all type of expression.

I think we can refactor and merge all related rule(include SimplifyNotExprRule) in the NormalizeExpressionRule in the future? @morrySnow @EmmyMiao87 @wangshuo128